### DISCUSSION TITLE
axios-es6-class@4.2.2" has incorrect peer dependency "axios@0.20.0

### DISCUSSION THREAD

**Discussion by DesAWSume** [axios/axios#5860] [Q&A]
### Describe the issue

Trying to update axios module but encounter incorrect peer dependency with axios-es6-class@4.2.2

Can anyone advise how to upgrade this safely without incorrect peer dependency? 

### Example Code

```js
Package.json
"dependencies": {
    "@typescript-eslint/eslint-plugin": "^6.1.0",
    "@typescript-eslint/parser": "^6.1.0",
    "axios": "^0.20.0",
    "axios-es6-class": "^4.2.2",
    "eslint": "^8.45.0",
    "jest": "^29.6.1",
    "nunjucks": "^3.2.2",
    "shx": "^0.3.2",
    "ts-jest": "^29.1.1",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.6"
  },
  "devDependencies": {},
```

### Expected behavior

update from 0.20.0 to 0.21.0 and above with no incorrect peer dependency

### Axios Version

0.20.0

### Adapter Version

HTTP

### Node.js Version

v18.16.0

### OS

13.4.1 (c) 

### Additional Library Versions

```bash
"@typescript-eslint/eslint-plugin": "^6.1.0",
    "@typescript-eslint/parser": "^6.1.0",
    "axios": "^0.20.0",
    "axios-es6-class": "^4.2.2",
    "eslint": "^8.45.0",
    "jest": "^29.6.1",
    "nunjucks": "^3.2.2",
    "shx": "^0.3.2",
    "ts-jest": "^29.1.1",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.6"
```

**Comment by jasonsaayman** [comment]
`axios-es6-class` is a third-party package, so its peer dependency range is controlled by that package, not by Axios.

If `axios-es6-class@4.2.2` declares a peer dependency on `axios@0.20.0`, your safe options are:

- Upgrade `axios-es6-class` if it has a newer version that supports the Axios version you want.
- Ask/submit a PR to `axios-es6-class` to widen its peer dependency range.
- Remove/replace `axios-es6-class` if it is no longer maintained.
- Use your package manager's peer-dependency override/resolution only after testing, because that bypasses the package author's compatibility declaration.

Also note that `axios@0.20.0` is very old. If you are upgrading Axios, consider moving to the latest `1.x` version and checking the migration guide rather than only `0.21.0`.
